Tools/Cli-LoRa-Device-Provisioning/IoTDeviceHelper.cs

```diff
@@ -0,0 +1,1113 @@
+                        isValid = false;
+                    }
+
+                    StatusConsole.WriteLine(MessageType.Warning, $"FCntUpStart is invalid/ignored for OTAA devices.");
+                }
+
+                // FCntDownStart
+                if (!string.IsNullOrEmpty(opts.FCntDownStart))
+                {
+                    if (!ValidationHelper.ValidateUIntTwinProperty(opts.FCntDownStart, 0, uint.MaxValue, out validationError))
+                    {
+                        StatusConsole.WriteLine(MessageType.Error, $"FCntDownStart {opts.FCntDownStart} is invalid: {validationError}.");
+                        isValid = false;
+                    }
+
+                    StatusConsole.WriteLine(MessageType.Warning, $"FCntDownStart is invalid/ignored for OTAA devices.");
+                }
+
+                // FCntResetCounter
+                if (!string.IsNullOrEmpty(opts.FCntResetCounter))
+                {
+                    if (!ValidationHelper.ValidateUIntTwinProperty(opts.FCntResetCounter, 0, uint.MaxValue, out validationError))
+                    {
+                        StatusConsole.WriteLine(MessageType.Error, $"FCntResetCounter {opts.FCntResetCounter} is invalid: {validationError}.");
+                        isValid = false;
+                    }
+
+                    StatusConsole.WriteLine(MessageType.Warning, $"FCntResetCounter is invalid/ignored for OTAA devices.");
+                }
+            }
+
+            // ******************************************************************************************
+            // // Shared, non optional device properties
+            // ******************************************************************************************
+
+            // SensorDecoder
+            if (!ValidationHelper.ValidateSensorDecoder(opts.SensorDecoder))
+            {
+                isValid = false;
+            }
+
+            // GatewayId
+            if (opts.GatewayId == null)
+            {
+                StatusConsole.WriteLine(MessageType.Error, $"GatewayId is missing.");
+                isValid = false;
+            }
+            else if (opts.GatewayId == string.Empty)
+            {
+                StatusConsole.WriteLine(MessageType.Info, $"GatewayId is empty. This is valid.");
+            }
+            else
+            {
+                StatusConsole.WriteLine(MessageType.Info, $"GatewayId {opts.GatewayId} is valid.");
+            }
+
+            // ******************************************************************************************
+            // // Shared, optional device properties
+            // ******************************************************************************************
+
+            // ClassType
+            if (!string.IsNullOrEmpty(opts.ClassType))
+            {
+                if (!(string.Equals(opts.ClassType, "A", StringComparison.InvariantCultureIgnoreCase) ||
```

Comment:
suggestion: add an array with "SupportedDeviceClasses and use that. Easier to extend and change if we need to add B to it.

Reply:
Changed to:

private string[] classTypes = { "A", "C" };

if (!Array.Exists(this.classTypes, classType => string.Equals(classType, opts.ClassType, StringComparison.OrdinalIgnoreCase)))

Reply:
Done.